src/Peering/Peering/Direct/SetAzureDirectPeeringConnectionCommand.cs

```diff
@@ -197,11 +197,11 @@ private PSDirectConnection UpdatePeeringOffer()
         /// </returns>
         private PSDirectConnection UpdateMD5Authentication()
         {
-            if (this.InputObject is PSDirectConnection inputObject)
+            if (this.InputObject.GetType().Equals(typeof(PSDirectConnection)))
```

Comment:
Why use direct type comparsion instead of `is`? Except for extremely rare cases, `is` is what you want.